dashboard-backend/pkg/fetchLog/fetchLog.go

```diff
@@ -0,0 +1,134 @@
+package fetchLog
+
+import (
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/Aman-Codes/e2e-dashboard-backend/pkg/constants"
+	"github.com/Aman-Codes/e2e-dashboard-backend/pkg/customErrors"
+	"github.com/Aman-Codes/e2e-dashboard-backend/pkg/env"
+	"github.com/Aman-Codes/e2e-dashboard-backend/pkg/unzip"
+	"github.com/Aman-Codes/e2e-dashboard-backend/pkg/utils"
+	"github.com/gin-gonic/gin"
+	"github.com/litmuschaos/litmus-go/pkg/log"
+)
+
+type LogsInput struct {
+	PipelineId string `json:"pipelineId" binding:"required"`
+	JobName    string `json:"jobName" binding:"required"`
+	StepNumber string `json:"stepNumber" binding:"required"`
+}
+
+func parseUrl(fullURLFile string) (string, error) {
+	log.Info("Start to fetch log")
+	fileURL, err := url.Parse(fullURLFile)
+	if err != nil {
+		log.Errorf("failed to parse URL, err %v", err)
+		return "", customErrors.InternalServerError()
```

Comment:
```suggestion
		return "", errors.Errorf( "failed to parse URL, err %v", customErrors.InternalServerError())
```
Please check this for other places as well

Reply:
Could you please tell which `errors` package are you referring to?
As using `log.Errorf` gives the below error
![image](https://user-images.githubusercontent.com/54680709/142163778-99a26553-e4fa-4cf9-bb38-0d554c516feb.png)

Reply:
okay will update it